sga/middleware.py

```diff
@@ -17,30 +17,38 @@ class SGAMiddleware(object):
     Middleware for processing incoming LTI requests
     """
     ADMIN_ROLES = ["Administrator", "Instructor"]
+    LTI_MIDDLEWARE_NOT_INSTALLED_MESSAGE = "LTI middleware not installed"
+    UNSUCCESSFUL_LTI_AUTHENTICATION_MESSAGE = "Bad or missing LTI credentials"
+    NO_CONTEXT_ID_MESSAGE = "No context_id in LTI parameters"
+    NO_RESOURCE_LINK_ID_MESSAGE = "No resource_link_id in LTI parameters"
+    REQUEST_USERNAME_FALSE_MESSAGE = "\"Request user's username\" must be set to True on this assignment."
 
     def process_request(self, request):  # pylint: disable=no-self-use
         """
         Processes incoming LTI requests
         """
         if not hasattr(request, "LTI"):
-            raise ImproperlyConfigured("LTI middleware not installed")
+            raise ImproperlyConfigured(self.LTI_MIDDLEWARE_NOT_INSTALLED_MESSAGE)
         if "course_roles" not in request.session:
             request.session["course_roles"] = {}
 
         if request.lti_initial_request:
```

Comment:
it looks like the rest of the function is under this `if` block: maybe you should `return` if the `request.lti_initial_request` is `False` and move the rest of the code on the left-er block.